### PR TITLE
Add view all button when more than 4 cards attached

### PR DESCRIPTION
**Pull request by FadhlanR**
Ticket: CS-6423

https://github.com/cardstack/boxel/assets/12637010/b42b41e2-9ed6-422e-89b9-bd33bb4cc32e

### CODE REVIEW

**Comment by habdelra** [comment]
isn't the ticket to show view all when there are _more_ than 4 cards? if so that means we show this on the 5th card, not the 4th card.

File: packages/matrix/tests/messages.spec.ts [hunk 1]
```diff
@@ -325,4 +325,32 @@ test.describe('Room messages', () => {
       },
     ]);
   });
+
+  test('displays view all pill if attached card more than 3', async ({ page }) => {
```
Comment (habdelra) [inline packages/matrix/tests/messages.spec.ts]: should be more than 4, not more than 3 right?

Comment (FadhlanR) [reply packages/matrix/tests/messages.spec.ts:329]: But in the description, it says "When more than 4 cards are attached to a message, show pills for the first 3 and then a "View all" pill".

Comment (FadhlanR) [reply packages/matrix/tests/messages.spec.ts]: My bad, I updated it here [3c8bbc7](https://github.com/cardstack/boxel/pull/1044/commits/3c8bbc7e97a75e848bd855462edd28a637298984).

File: packages/host/app/components/ai-assistant/card-picker/index.gts [hunk 2]
```diff
@@ -127,9 +148,32 @@ export default class AiAssistantCardPicker extends Component<Signature> {
...
+
+  @action
+  private shouldDisplayCard(card: CardDef, index: number): boolean {
+    if (
+      this.isViewAllAttachedCards ||
+      this.cardsToDisplay.length <= MAX_CARDS_TO_DISPLAY
+    ) {
+      return !!card.id;
+    } else {
+      // Displays the first three cards
```
Comment (jurgenwerk) [inline packages/host/app/components/ai-assistant/card-picker/index.gts]: Comment outdated?

File: packages/host/app/components/ai-assistant/card-picker/index.gts [hunk 3]
```diff
@@ -69,6 +71,25 @@ export default class AiAssistantCardPicker extends Component<Signature> {
...
+        (and
+          (gt this.cardsToDisplay.length MAX_CARDS_TO_DISPLAY)
+          (not this.isViewAllAttachedCards)
+        )
+      }}
+        <Pill
+          @inert={{true}}
+          class='card-pill view-all'
+          data-test-view-all
+          {{on 'click' this.toogleViewAllAttachedCards}}
```
Comment (jurgenwerk) [inline packages/host/app/components/ai-assistant/card-picker/index.gts]:
```suggestion
          {{on 'click' this.toggleViewAllAttachedCards}}
```

File: packages/host/app/components/ai-assistant/card-picker/index.gts [hunk 4]
```diff
@@ -127,9 +148,32 @@ export default class AiAssistantCardPicker extends Component<Signature> {
...
 
+  @tracked isViewAllAttachedCards = false;
+
+  @action
+  private toogleViewAllAttachedCards() {
+    this.isViewAllAttachedCards = !this.isViewAllAttachedCards;
+  }
+
+  @action
+  private shouldDisplayCard(card: CardDef, index: number): boolean {
```
Comment (jurgenwerk) [inline packages/host/app/components/ai-assistant/card-picker/index.gts]: Maybe `isCardDisplayed` is more concise

File: packages/host/app/components/ai-assistant/card-picker/index.gts [hunk 5]
```diff
@@ -127,9 +148,32 @@ export default class AiAssistantCardPicker extends Component<Signature> {
       .is-autoattached {
         border-style: dashed;
       }
+      .view-all {
+        cursor: pointer;
```
Comment (jurgenwerk) [inline packages/host/app/components/ai-assistant/card-picker/index.gts:152]: Hm, isn't this pill a button tag, which should have pointer enabled by default?

**Review by jurgenwerk — APPROVED** [review]

File: packages/host/app/components/ai-assistant/card-picker/index.gts (hunk 5 above)
Comment (FadhlanR) [reply packages/host/app/components/ai-assistant/card-picker/index.gts:152]: No it isn't, I reused `Pill` component for this.

**Pull request merged**